libqtile/widget/window_count.py

```diff
@@ -55,21 +58,20 @@ def _setup_hooks(self):
 
     def _wincount(self, *args):
         try:
-            self._count = len(self.qtile.current_group.windows)
+            self._count = len(self.bar.screen.group.windows)
         except AttributeError:
             self._count = 0
 
         self.update(self.text_format.format(num=self._count))
 
     def _win_killed(self, window):
         try:
-            self._count = len(self.qtile.current_group.windows)
+            self._count = len(self.bar.screen.group.windows)
+            if window.group == self.bar.screen.group:
```

Comment:
Which one of these gives rise to an `AttributeError`?

Reply:
The `if window.group == self.bar.screen.group:` initially gave an error in the `test_attribute_errors` testcase.
But as you can see I already moved it into the try/catch block, so now it works as expected. When there is no bar, screen or group the number of windows is 0 and therefore self._count gets set to 0.

So now you only have to write some new tests for the multi monitor setup.

Reply:
Ah, yes. `window` can be `None` so `window.group` raises the exception.